internal/pkg/networkscanner/portdiscovery/portdiscovery.go

```diff
@@ -0,0 +1,267 @@
+package portdiscovery
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/kubescape/kubescape-network-scanner/internal/pkg/networkscanner"
+)
+
+type ScanConfig struct {
+	Targets []networkscanner.TargetDescription
+	Ports   []int
+	Timeout time.Duration
+	TcpOnly bool
+	UdpOnly bool
+}
+
+type ScanTarget struct {
+	Hostname string
+	IPStart  net.IP
+}
+
+// Struct defining the result of the network scanner
+
+func parseArgs() (*ScanConfig, error) {
```

Comment:
Looks like part of the CLI (should be moved there)

Reply:
Can we work on the Cli on a separate PR?
Like i've made changes according to the go styling guide like package name should be the directory names. 
So i've change the path to `internal/pkg/networkscanner/` from `internal/pkg/network-scanner/`.

Reply:
Sure, just note this